downburst/template.py

```diff
@@ -123,6 +146,11 @@ def domain(
     if raw:
         type = 'raw'
     (devices,) = tree.xpath('/domain/devices')
+    emulator = devices.find('emulator')
+    if emulator is not None:
+        emulator.text = get_emulator_path()
+    else:
+        etree.SubElement(devices, 'emulator').text = get_emulator_path()
```

Comment:
IIUC, you are monkey-patching the XML, to fix the "emulator" element? if that's the case, could you add a comment here?

Reply:
this looks to me not like a fix, not sure this is monkey-patching since we work with the template, however, could you give me some comment example you want me to add, or if this is more vulnerable manipulation is it worth to add log.warning message?

Reply:
what i don't understand is why you need to check if emulator is `None`. i think a template is always predefined, and we should know if the element is available or not beforehand. if that's the case, why bothering checking it?

Reply:
Added this in case someone removes the emulator from the template, for example, we decide to remove it from there, since it is always overridden, like it done for disks etc.
So I'm happy to do anything from these. So what do you prefer from those three, remove this extra check, remove 'emulator' from the template, leave it as is and warning/debug/comment?

Reply:
thanks @kshtsk , i felt rich when facing all these options!

okay, so if template is allowed to be changed, i'd go with option 3: "leave it as is and warning/debug/comment".